java-engine/src/main/java/io/getunleash/engine/FeatureDef.java

```diff
@@ -0,0 +1,34 @@
+package io.getunleash.engine;
+
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+class FeatureDef {
+    private final String name;
+    private final Optional<String> type;
+    private final String project;
+
+    @JsonCreator
+    FeatureDef(
+            @JsonProperty("name") String name,
+            @JsonProperty("type") String featureType,
+            @JsonProperty("project") String project) {
+        this.name = name;
+        this.project = project;
+        this.type = Optional.ofNullable(featureType);
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Optional<String> getType() {
```

Comment:
Why an optional when your constructor takes a string?

Reply:
So I'd like the constructor to take an Optional too but apparently Jackson hates that so I figured me wanting that was wrong for some reason.